store/instance.go

```diff
@@ -35,6 +35,13 @@ type instanceRaw struct {
 	Port          string
 }
 
+// InstanceCountMetric is the API message for instance count metric, used by CountInstanceGroupByEngineAndEnvironmentID
+type InstanceCountMetric struct {
```

Comment:
ideally, this should be put together with the metric name definition so that one can see all definitions in a single place.

Reply:
So we have to:

1. Put the metric API messages back to the `api` package, that's what we did in the previous commit.
2. Put the `CountInstanceGroupByEngineAndEnvironmentID` to the `collector` package, but it's not a good way as this need to access the `db` in `store` from `collector`

We cannot just put the `InstanceCountMetric` into the `collector` package as this will cause the package reference circle.

Reply:
I think we can just put the `InstanceCountMetric` into the `api/metric.go` file?

Reply:
Updated, also add the `IssueCountMetric` for issue